staging/src/k8s.io/client-go/examples/controller-ref-manager/main.go

```diff
@@ -0,0 +1,166 @@
+	// _ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+
+	"k8s.io/client-go/examples/controller-ref-manager/controller"
+)
+
+var (
+	productionLabel         = map[string]string{"type": "production", "app": "demo"}
+	productionLabelSelector = labels.Set{"type": "production"}.AsSelector()
+)
+
+// RefManager normally used in controllers sync loop to adopt/release resources.
+// See https://github.com/kubernetes/community/blob/master/contributors/design-proposals/api-machinery/controller-ref.md
+// for more details.
+func main() {
+	kubeconfig := flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.Parse()
+	if *kubeconfig == "" {
+		panic("-kubeconfig not specified")
+	}
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	myController := controller.MyController{ObjectMeta: metav1.ObjectMeta{Name: "demo"}}
+	myController.UID = types.UID("123")
+
+
+	deployment := &apps.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "demo-deployment",
+			Labels: productionLabel,
+		},
+		Spec: apps.DeploymentSpec{
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": "demo"}},
+			Replicas: int32Ptr(2),
+			Template: apiv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "demo",
+					},
+				},
+				Spec: apiv1.PodSpec{
+					Containers: []apiv1.Container{
+						{
+							Name:  "web",
+							Image: "nginx:1.13",
+							Ports: []apiv1.ContainerPort{
+								{
+									Name:          "http",
+									Protocol:      apiv1.ProtocolTCP,
+									ContainerPort: 80,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Create Deployment without OwnerReferences
+	fmt.Println("Creating deployment...")
+	deploymentsClient := clientset.AppsV1beta2().Deployments(apiv1.NamespaceDefault)
+	result, err := deploymentsClient.Create(deployment)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("Created deployment %q.\n", result.GetObjectMeta().GetName())
+	fmt.Printf("You can check OwnerReferences using kubectl: \"kubectl get deployment %s -o yaml\" \n", result.GetObjectMeta().GetName())
+	prompt()
+
+
+	// If any adoptions are attempted, we should first recheck for deletion with
+	// an uncached quorum (see #42639).
+	canAdoptFunc := basecontroller.RecheckDeletionTimestamp(func() (metav1.Object, error) {
+		fresh, err := clientset.AppsV1beta1().Deployments(result.Namespace).Get(result.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		if fresh.UID != result.UID {
+			return nil, fmt.Errorf("original Deployment %v/%v is gone: got uid %v, wanted %v",
+				result.Namespace, result.Name, fresh.UID, result.UID)
+		}
+		return fresh, nil
+	})
+
+	gvk := schema.GroupVersionKind{Group: "test", Version: "v1beta2", Kind: "test"}
```

Comment:
missing a comment why an arbitrary gvk can be used here. Kind is usually upper-case, group should better be something like `examples.client-go.k8s.io`.